### ISSUE TITLE
or_fun_call warns on enum variant construction.

### ISSUE THREAD

**Issue by Stebalien** [Manishearth/rust-clippy#917]
I have the following line in my project:

```rust
self.write_to_fmt(&mut Adapter(f)).or(Err(fmt::Error))
```

`Err(fmt::Error)` is a constant but clippy complains that I'm calling a function.

**Comment by mcarton** [comment]
`Err` (as all parametered enum variants) is also a function. But Clippy should not warn there indeed.

**Comment by llogiq** [comment]
I've seen other instances where functions that were basically constructors (and fully inclined) were suggested to use lazily. Alas, we don't have a measurement for 'this is expensive' in clippy. Perhaps the message should reflect this.

**Comment by Stebalien** [comment]
Thanks!

**Comment by llogiq** [comment]
Thank you for letting us know!
